core/arch/arm/kernel/spin_lock_debug.c

```diff
@@ -25,20 +25,38 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef TZ_PROC_H
-#define TZ_PROC_H
+#include <assert.h>
+#include <kernel/spinlock.h>
+#include "thread_private.h"
 
-#define SPINLOCK_LOCK       1
-#define SPINLOCK_UNLOCK     0
+void spinlock_count_incr(void)
+{
+	struct thread_core_local *l = thread_get_core_local();
 
-#ifndef ASM
-void cpu_spin_lock(unsigned int *lock);
-unsigned int cpu_spin_trylock(unsigned int *lock);
-void cpu_spin_unlock(unsigned int *lock);
+	l->locked_count++;
```

Comment:
An `assert(l->locked_count)` after the incrementation could catch overflow. unlikely to happen but it won't cost, and... for consistency.